gcsfs/core.py

```diff
@@ -893,17 +897,33 @@ def flush(self, force=False):
             raise ValueError('Flush on a file not in write mode')
         if self.closed:
             raise ValueError('Flush on closed file')
+
         if self.buffer.tell() == 0 and not force:
             # no data in the buffer to write
             return
+        if self.buffer.tell() < GCS_MIN_BLOCK_SIZE and not force:
+            warnings.warn(
+                "GCSFile.flush(force=False) with buffer size (%s) below minimum GCS chunk size (2 ** 18), "
+                "skipping block upload." % self.buffer.tell()
+            )
+            return
+
         if force and self.forced:
             raise ValueError("Force flush cannot be called more than once")
-        if force and not self.offset and self.buffer.tell() <= 5 * 2**20:
-            self._simple_upload()
-        elif not self.offset:
-            self._initiate_upload()
+
+
+        if not self.offset:
+            if force and self.buffer.tell() <= self.blocksize:
```

Comment:
Actually, I think we should should do this for any file smaller than DEFAULT_BLOCK_SIZE.
Along the same vane, maybe there's no need to call `_initiate_upload` until we've accumulated more than DEFAULT_BLOCK_SIZE, and simple upload might still happen.

Reply:
I'm not sure I understand the rational for overriding the user-specified block size? Isn't the intended function of that feature to allow explicit control over cutoffs for an upload request, whether simple or chunked?

Agreed on the second point.

Reply:
Hm, OK maybe you're right. Honestly, I don't expect anyone to use a non-standard block size for writing anyway.